components/core/src/clp/GlobalMetadataDBConfig.cpp

```diff
@@ -1,110 +1,99 @@
 #include "GlobalMetadataDBConfig.hpp"
 
+#include <cstdlib>
+#include <stdexcept>
+
 #include <fmt/core.h>
-#include <yaml-cpp/yaml.h>
 
-using std::exception;
 using std::invalid_argument;
 using std::string;
 
-static exception get_yaml_missing_key_exception(string const& key_name) {
-    throw invalid_argument(fmt::format("Missing key '{}'", key_name));
-}
+namespace clp {
+// Constants
+constexpr int cMinPort{1};
+constexpr int cMaxPort{65'535};
+
+auto operator>>(std::istream& in, GlobalMetadataDBConfig::MetadataDBType& metadata_db_type)
+        -> std::istream& {
+    string db_type_string;
+    in >> db_type_string;
+
+    if ("sqlite" == db_type_string) {
+        metadata_db_type = GlobalMetadataDBConfig::MetadataDBType::SQLite;
+    } else if ("mysql" == db_type_string) {
+        metadata_db_type = GlobalMetadataDBConfig::MetadataDBType::MySQL;
+    } else {
+        throw invalid_argument("Unknown database type: " + db_type_string);
+    }
 
-static exception
-get_yaml_unconvertable_value_exception(string const& key_name, string const& destination_type) {
-    throw invalid_argument(
-            fmt::format("'{}' could not be converted to type '{}'", key_name, destination_type)
-    );
+    return in;
 }
 
-namespace clp {
-void GlobalMetadataDBConfig::parse_config_file(string const& config_file_path) {
-    YAML::Node config = YAML::LoadFile(config_file_path);
+auto GlobalMetadataDBConfig::add_command_line_options(
+        boost::program_options::options_description& options_description
+) -> void {
+    options_description.add_options()
+            ("db-type",
+             boost::program_options::value<MetadataDBType>(&m_metadata_db_type)->default_value(cDefaultMetadataDbType, cDefaultMetadataDbTypeName.data()),
+             "Database type [sqlite | mysql]")
+            ("db-host",
+             boost::program_options::value<string>(&m_metadata_db_host)->default_value(cDefaultMetadataDbHost.data()),
```

Comment:
I feel that if have already default constructed the member variables, we can use them for the default_values(). such that if we want to update the default values we don't need to update two places.

there are some examples like https://github.com/y-scope/clp/blob/d21506b3be534dbb7da0d1aa112c758ceb96208f/components/core/src/clp/clp/CommandLineArguments.cpp#L237

Reply:
> if have already default constructed the member variables, we can use them for the default_values()
i see your point. that said, what do you think about my rationale to use the `c` constants instead of the member variables? https://github.com/y-scope/clp/pull/1148#discussion_r2248861853

> if we want to update the default values we don't need to update two places.

not sure if i get this point. what are the two places that we need to update when we want to update the default values?

Reply:
Just a reminder, this request seems to be not either addressed or rejected @junhaoliao

Reply:
I think i get your point now. I still think it’s the best if we can reference the default values directly in the cli arguments’s registration, unless we prohibit this method to be called more than once. In such case, would it be better if we remove the member fields’ initialisations at their declarations?